schema/validator.go

```diff
@@ -88,11 +89,44 @@ func (v Validator) Validate(src io.Reader) error {
 
 type unimplemented string
 
-func (v unimplemented) Validate(src io.Reader) error {
+func (v unimplemented) Validate(src io.Reader, strict bool) error {
 	return fmt.Errorf("%s: unimplemented", v)
 }
 
-func validateManifestDescendants(r io.Reader) error {
+func validateMediaTypeInManifestList(r io.Reader, strict bool) error {
+	header := v1.ManifestList{}
+
+	buf, err := ioutil.ReadAll(r)
+	if err != nil {
+		return errors.Wrapf(err, "error reading the io stream")
+	}
+
+	err = json.Unmarshal(buf, &header)
+	if err != nil {
+		return errors.Wrap(err, "manifest list format mismatch")
+	}
+
+	if header.MediaType != string(v1.MediaTypeImageManifestList) {
+		msg := fmt.Sprintf("manifest list has an unknown media type: %s", header.MediaType)
+		if strict {
+			return fmt.Errorf("%s", msg)
+		}
+		fmt.Printf("warning: %s\n", msg)
+	}
+
+	for _, manifest := range header.Manifests {
+		if manifest.MediaType != string(v1.MediaTypeImageManifest) {
+			msg := fmt.Sprintf("manifest %s has an unknown media type: %s", manifest.Digest, manifest.MediaType)
+			if strict {
+				return fmt.Errorf("%s", msg)
+			}
+			fmt.Printf("warning: %s\n", msg)
```

Comment:
I think we should stick to printing warnings for this PR.

In follow-up work we can add a validation argument which can hold a test-harness handle (e.g. a more generic form of the [`harness *tap.T`][1] I mocked up in opencontainers/image-tools@4f0b3c1b).  Then these test functions can register errors and warnings without worrying about how the results are being collected or displayed.  But we don't need to bite all of that off in one chunk ;).

[1]: https://github.com/opencontainers/image-tools/commit/4f0b3c1bae97d212680dc73ee8b35b6e47b8a8c6#diff-8351286f57eb81e245c9c99c07f3b34fR28